Sources/NIOHTTP2/HTTP2ChannelHandler+InlineStreamMultiplexer.swift

```diff
@@ -239,7 +239,7 @@ extension NIOHTTP2Handler {
         }
 
         /// Create a stream channel initialized with the provided closure
-        public func createStreamChannel<OutboundStreamOutput>(_ initializer: @escaping NIOChannelInitializerWithOutput<OutboundStreamOutput>) async throws -> OutboundStreamOutput {
+        public func createStreamChannel<OutboundStreamOutput: Sendable>(_ initializer: @escaping NIOChannelInitializerWithOutput<OutboundStreamOutput>) async throws -> OutboundStreamOutput {
```

Comment:
I think `OutboundStreamOutput` is unnecessarily verbose here: `Output` is fine.

Reply:
Agreed. `Output` should be enough.

Reply:
I personally still think we should call this method `createOutboundStream` so that it aligns with the `inboundStreamInitializer`.

Reply:
This name aligns with the equivalent method on the non-async multiplexer, I think that alignment is more important.

Reply:
I personally don't think that is more important because this here is the chance for a new name. I recently saw somebody giving feedback on exactly this API that they are confused what initializer runs when.

Reply:
I think `createOutboundStream` is misleading: it sounds like it's a unidirectional stream to me, which it isn't. (FWIW I have the same issue with our inbound stream initializers.)